pkg/probe/turbo_probe.go

```diff
@@ -70,38 +73,32 @@ func newTurboProbe(probeConf *ProbeConfig) (*TurboProbe, error) {
 
 	myProbe := &TurboProbe{
 		ProbeConfiguration: probeConf,
-		DiscoveryClientMap: make(map[string]*TargetDiscoveryAgent),
+		DiscoveryClient:    nil, // DiscoveryClient is originally empty. Should be initialized later.
 		RegistrationClient: NewProbeRegistrator(),
+		TargetsToAdd:       map[string]bool{},
 	}
 
 	glog.V(2).Infof("[NewTurboProbe] Created TurboProbe: %v", myProbe)
 	return myProbe, nil
 }
 
-func (theProbe *TurboProbe) getDiscoveryClient(targetIdentifier string) TurboDiscoveryClient {
-	target, exists := theProbe.DiscoveryClientMap[targetIdentifier]
-
-	if !exists {
-		glog.Errorf("[GetTurboDiscoveryClient] Cannot find Target for address: %s", targetIdentifier)
+func (theProbe *TurboProbe) getDiscoveryClient() TurboDiscoveryClient {
+	if theProbe.DiscoveryClient == nil {
+		glog.Errorf("[GetTurboDiscoveryClient] No DiscoveryAgent configured.")
 		return nil
 	}
-	return target.TurboDiscoveryClient
+
+	return theProbe.DiscoveryClient.TurboDiscoveryClient
```

Comment:
Shall we check nil for `theProbe.DiscoveryClient.TurboDiscoveryClient`?

Reply:
I'm removing this function according to your suggestion above, but for my own understanding, what would be the advantage of checking for a nil `theProbe.DiscoveryClient.TurboDiscoveryClient` here since all we're doing is returning the value? To log an error message? Isn't it up to the client of the method to check for nil before use?